support/testsupport/src/main/java/com/hotels/styx/support/Latch.java

```diff
@@ -0,0 +1,57 @@
+/*
+  Copyright (C) 2013-2018 Expedia Inc.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+package com.hotels.styx.support;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import static com.google.common.base.Throwables.propagate;
+import static java.lang.Thread.currentThread;
+
+/**
+ * Wrapper for CountDownLatch to make it more convenient to use.
+ */
+public class Latch {
```

Comment:
As per our unwritten naming conventions, utility classes names like this would have their name pluralised.

That is, the class name should be `Latches`.

Reply:
It's not a utility class. Utility classes have only static methods. This is a normal class.

Reply:
Right of course. Now I see that. I should have seen that straight away. 

But why to implement these await methods as a composed class? 

By making it a public test support class you are promoting its use in other tests. But then again the `Latch` class is not type compatible with `CountDownLatch`. Sooner or later some parts of the codebase will be using `Latch` while other parts stick to `CountDownLatch`. After that we will need to bridge the two classes at the interface level.

But all we really want is to re-propagate checked exceptions as runtime exceptions in `await` method.

For these reasons, please convert this to a utility class with `await` as a static utility method.

Reply:
I'll look into this.